spot-contracts/contracts/FeePolicy.sol

```diff
@@ -227,32 +265,78 @@ contract FeePolicy is IFeePolicy, OwnableUpgradeable {
         return vaultDeploymentFee;
     }
 
+    /// @inheritdoc IFeePolicy
+    function computeUnderlyingToPerpSwapFeePercs(uint256 valueIn) external override returns (uint256, uint256) {
+        // When user swaps underlying for vault's perps -> perps are minted by the vault
+        // Similar to perp mint fees, here too we use the "post"-minting subscription state.
+        uint256 currentPerpTVL = perp.getTVL();
+        uint256 currentVaultTVL = vault.getTVL();
+        IFeePolicy.SubscriptionState memory postMintState = computeSubscriptionState(
+            perp.getDepositBond(),
+            currentPerpTVL + valueIn,
+            currentVaultTVL
+        );
+
+        // If minting leaves the vault under-subscribed, swapping by minting perps is NOT allowed.
+        if (postMintState.normalizedSubscriptionRatio <= 1) {
+            return (0, ONE);
+        }
+
+        // When the system is over-subscribed, we charge no perp mint fee.
+        uint256 vaultFeePerc = vaultSwapFeePerc[
+            keccak256(abi.encodePacked(address(vault.underlying()), address(perp)))
+        ];
+        return (0, vaultFeePerc);
+    }
+
+    /// @inheritdoc IFeePolicy
+    function computePerpToUnderlyingSwapFeePercs() external override returns (uint256, uint256) {
+        // When user swaps perps for vault's underlying -> perps are redeemed by the vault
+        IFeePolicy.SubscriptionState memory s = computeSubscriptionState();
+
+        // Split the fees between the perp and vault systems
+        uint256 vaultFeePerc = vaultSwapFeePerc[
+            keccak256(abi.encodePacked(address(perp), address(vault.underlying())))
+        ];
+        // When the system is over-subscribed, we charge no perp burn fee.
```

Comment:
Is this backwards? I thought...
undersubscribed -> no burn fee
oversubscribed -> burn fee

Reply:
Good catch, When the system is undersubscribed no perp redemption fee (we make redemptions easy); when the system is oversubscribed we charge a perp redemption fee.